modules/agar/src/test/ts/browser/FocusToolsTest.ts

```diff
@@ -1,104 +1,116 @@
 import { Assert, UnitTest } from '@ephox/bedrock-client';
-import { SugarElement, Value } from '@ephox/sugar';
+import { SugarElement, SugarShadowDom, Value } from '@ephox/sugar';
 import * as Assertions from 'ephox/agar/api/Assertions';
 import { Chain } from 'ephox/agar/api/Chain';
 import * as FocusTools from 'ephox/agar/api/FocusTools';
+import * as GeneralSteps from 'ephox/agar/api/GeneralSteps';
 import * as Guard from 'ephox/agar/api/Guard';
+import * as Log from 'ephox/agar/api/Log';
 import { Pipeline } from 'ephox/agar/api/Pipeline';
 import { Step } from 'ephox/agar/api/Step';
 import * as DomContainers from 'ephox/agar/test/DomContainers';
 
 UnitTest.asynctest('FocusToolsTest', (success, failure) => {
 
-  const doc = SugarElement.fromDom(document);
-  const docNode = SugarElement.fromDom(document.documentElement);
-
-  Pipeline.async({}, [
-    DomContainers.mSetup,
-    Step.log('cat1'),
-    FocusTools.sSetFocus('Focusing body', docNode, 'body'),
+  const sTestFocusTools = (doc: SugarElement, docNode: SugarElement) =>
+    GeneralSteps.sequence([
+      FocusTools.sSetFocus('Focusing div', docNode, 'div[test-id]'),
+      FocusTools.sIsOnSelector('Should be on div[test-id]', doc, 'div[test-id]'),
+      FocusTools.sSetFocus('Focusing input', docNode, 'div[test-id] input'),
+      FocusTools.sIsOnSelector('Should be on div[test-id] input', doc, 'div[test-id] input'),
+      FocusTools.sSetActiveValue(doc, 'new value'),
 
-    Step.log('cat2'),
-    FocusTools.sIsOnSelector('Should be on body', doc, 'body'),
-    FocusTools.sSetFocus('Focusing div', docNode, 'div[test-id]'),
-    FocusTools.sIsOnSelector('Should be on div[test-id]', doc, 'div[test-id]'),
-    FocusTools.sSetFocus('Focusing input', docNode, 'div[test-id] input'),
-    FocusTools.sIsOnSelector('Should be on div[test-id] input', doc, 'div[test-id] input'),
-    FocusTools.sSetActiveValue(doc, 'new value'),
+      Chain.asStep(doc, [
+        FocusTools.cGetFocused,
+        Chain.control(
+          Chain.on((active, next, die, logs) => {
```

Comment:
Should that be a `Chain.op` or maybe even a `cAssertEq`?

Reply:
Unrelated to this change, but no that wouldn't work as it needs to feed through the previous value when calling next.

Reply:
Actually on a second look it probably could be `Chain.op`. Still unrelated to this change though.